integration_tests/src/main/python/hash_aggregate_test.py

```diff
@@ -335,12 +348,19 @@ def test_hash_reduction_decimal_overflow_sum(precision):
         # some optimizations are conspiring against us.
         conf = {'spark.rapids.sql.batchSizeBytes': '128m'})
 
-@pytest.mark.parametrize('data_gen', [_longs_with_nulls], ids=idfn)
+@pytest.mark.parametrize('data_gen', [_grpkey_nested_structs_with_array_basic_child,  _longs_with_nulls] + _grpkey_list_with_non_nested_children, ids=idfn)
 def test_hash_grpby_sum_count_action(data_gen):
     assert_gpu_and_cpu_row_counts_equal(
         lambda spark: gen_df(spark, data_gen, length=100).groupby('a').agg(f.sum('b'))
     )
 
+@allow_non_gpu("ShuffleExchangeExec", "HashAggregateExec")
+@pytest.mark.parametrize('data_gen', [_grpkey_nested_structs_with_array_child], ids=idfn)
+def test_hash_grpby_sum_count_action_fallback(data_gen):
```

Comment:
The commit comment says this is for testing shuffle exec fallback, but that's not really what this does.  This is more about testing hash aggregate fallback (and arguably is a duplicate of the existing `test_hash_agg_with_struct_of_array_fallback`).  The shuffle is falling back because both sides are also falling back, and it is inefficient to shuffle on the GPU to/from CPU exec nodes.

To have a test more focused on shuffle, it should be in repart_test.py and use something like `repartition` to force a shuffle.  See other tests in repart_test.py for examples.  Make sure the test passes with supported types and falls back for unsupported types.

Reply:
So Spark is deciding not to Shuffle this. I realized something was off about this right after pushing this. I am not seeing it hit the `wrapPart` method which was confusing me, thank you for clarifying as to why it's not replacing the ShuffleExec when it should.

Reply:
> thank you for clarifying as to why it's not replacing the ShuffleExec when it should.

But that's just it -- it should *not* replace the shuffle even if it wanted to.  Shuffling with arrays-of-struct as the partitioning key should not be supported by ShuffleExec just like we cannot support grouping with arrays-of-struct.   I was wrong above, `repartition` is not what we want here because that doesn't involve partitioning on any particular column.  We need to test partitioning on array-of-struct keys which s not supported.  Not sure we can easily test this at the integration test level that we're doing this properly, as I think we would need some way for Spark to hash partition on array-of-struct into GPU operations that support that but the shuffle does not.  But if we can do an operation that needs hash partitioning of array-of-struct (like groupby) then we should be able to hash partition it as well.

Reply:
I have added an integration test. I modified a repart_test by adding Array of structs as a type and then just refactored that into a separate test. I know you mentioned that repartition won't work but I wanted to run this by you to see if this does